pycheribuild/projects/cross/cheribsd.py

```diff
@@ -68,6 +71,333 @@ def _clear_dangerous_make_env_vars():
             del os.environ[k]
 
 
+class KernelABI(Enum):
+    NOCHERI = "no-cheri"
+    HYBRID = "hybrid"
+    PURECAP = "purecap"
+
+
+class ConfigPlatform(Enum):
+    QEMU = "qemu"
+    FVP = "fvp"
+    GFE = "gfe"
+    BERI = "cheri-beri"
+
+    @classmethod
+    def fpga_platforms(cls) -> set:
+        return {cls.BERI, cls.GFE}
+
+
+class CheriBSDConfig:
+    """
+    Cheribuild configuration descriptor for a CheriBSD kernel configuration file
+    """
+
+    def __init__(self, kernconf: str, platform: ConfigPlatform, **kwargs):
+        self.kernconf = kernconf
+        self.platform = {platform} if type(platform) != set else platform
+        self.kABI = kwargs.get("kABI", KernelABI.NOCHERI)
+        self.default = kwargs.get("default", False)
+        self.caprevoke = kwargs.get("caprevoke", False)
+        self.mfsroot = kwargs.get("mfsroot", False)
+        self.debug = kwargs.get("debug", False)
+        self.benchmark = kwargs.get("benchmark", False)
+        self.fuzzing = kwargs.get("fuzzing", False)
+        self.fett = kwargs.get("fett", False)
```

Comment:
```suggestion
    def __init__(self, kernconf: str, platform: ConfigPlatform, kABI=KernelABI.NOCHERI, default=False, ...):
        self.kernconf = kernconf
        self.platform = {platform} if type(platform) != set else platform
        self.kABI = kABI
        self.default = default
        ...
```
Is there a reason we can't use default arguments here?

Reply:
Not really, was easier to add new flags on the fly but perhaps less readable.